src/cf.data.pre

```diff
@@ -1486,6 +1486,13 @@ IF USE_OPENSSL
 	acl aclname ssl_error errorname
 	  # match against SSL certificate validation error [fast]
 	  #
+	  # When used with sslproxy_cert_error, this ACL tests a single
+	  # certificate validation error currently being evaluated by that
+	  # directive. When used with slproxy_cert_sign or sslproxy_cert_adapt,
+	  # the ACL tests all past certificate validation errors associated with
+	  # the current Squid-to-server connection (attempt). This ACL is not yet
+	  # supported for use with other directives.
+	  #
```

Comment:
> PR description: This change updates Squid documentation to reflect the above findings. Unfortunately, it is not feasible to remove this unwanted directive "sensitivity" without deprecating ssl_error and adding two new ACL types: Selecting any one of the two definitions above may seriously (and silently) break existing configurations: ...

I think it is best for this PR to focus on the bug fixes, but I cannot think of a better long-term approach than to deprecate and split `ssl_error` ACL, so that each of the two use cases (last/current-error and all-seen-errors) is covered by a dedicated ACL. Please let me know if you see a better solution.

Reply:
@chtsanti came up with a better long-term solution: Instead of splitting `ssl_error` into two ACLs as described in my comment above, we should provide a new _directive_ that tests whether the last/current error should be honored:

1. Deprecate/rename the `ssl_errors` ACL to something like `tls_errors` or `tls::errors` (to make sure that admins currently relying on its "sometimes this matches just the current validation error" semantics stop doing that) but keep it a single ACL that can be used (and has the same meaning) in any already supported context.

2. Deprecate `sslproxy_cert_error` in favor of the new `sslproxy_certificate_validation_error` or a similar directive that, by its very structure/syntax is focuses on handling just one error at a time:

```
sslproxy_certificate_validation_error <error-name> allow|deny acl...
```

Again, these plans do not affect the proposed bug fix (i.e. this PR code changes). I am just sharing ideas on how to solve the configuration problems identified while working on this PR.